staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/etcd.go

```diff
@@ -41,19 +39,19 @@ type CustomResourceStorage struct {
 	Scale          *ScaleREST
 }
 
-func NewStorage(resource schema.GroupResource, listKind schema.GroupVersionKind, strategy customResourceStrategy, optsGetter generic.RESTOptionsGetter, status *apiextensions.CustomResourceSubResourceStatus, scale *apiextensions.CustomResourceSubResourceScale) CustomResourceStorage {
+func NewStorage(resource schema.GroupResource, listKind schema.GroupVersionKind, strategy customResourceStrategy, optsGetter generic.RESTOptionsGetter) CustomResourceStorage {
 	customResourceREST, customResourceStatusREST := newREST(resource, listKind, strategy, optsGetter)
 	customResourceRegistry := NewRegistry(customResourceREST)
 
 	s := CustomResourceStorage{
 		CustomResource: customResourceREST,
 	}
 
-	if status != nil {
+	if strategy.validator.status != nil {
```

Comment:
odd that we look into the validator for this

Reply:
we need `scale` for validation (validating the json paths). So it needs to be in the validator to be accessible in `ValidateCreate` and `ValidateUpdate`. Thought it would be symmetrical to keep status there as well (though it seems somewhat counter-intuitive).

Reply:
would prefer to have status and scale in the strategy. It's strange to access the validator from here.

Reply:
Ok, can do.